pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -825,3 +827,19 @@ func getBaseosUpdateCounter(ctx *zedagentContext) uint32 {
 	status := m.(types.BaseOSMgrStatus)
 	return status.CurrentRetryUpdateCounter
 }
+
+func getState(ctx *zedagentContext) info.ZDeviceState {
+	if ctx.rebootCmd || ctx.deviceReboot {
+		return info.ZDeviceState_ZDEVICE_STATE_REBOOTING
+	}
+	if ctx.getconfigCtx != nil && ctx.getconfigCtx.updateInprogress {
+		return info.ZDeviceState_ZDEVICE_STATE_BASEOS_UPDATING
+	}
+	if ctx.maintenanceMode {
+		return info.ZDeviceState_ZDEVICE_STATE_MAINTENANCE_MODE
+	}
```

Comment:
The order of these seems off. If maintenanceMode is set I think that takes precedence over others.

Does updateInprogress remain set until we have completed the testing/inprogress of the new image?  We should report ZDEVICE_STATE_BASEOS_UPDATING from the start of the update until the testing is done.

BASEOS_UPDATING should take precedence over ZDEVICE_STATE_REBOOTING.

Reply:
> The order of these seems off. If maintenanceMode is set I think that takes precedence over others.

So we will miss other states in this case, well, it is not common-used state, so, why not, done.
 
> Does updateInprogress remain set until we have completed the testing/inprogress of the new image? We should report ZDEVICE_STATE_BASEOS_UPDATING from the start of the update until the testing is done.

updateInprogress indicates inprogress, so, I add checking for updating state of other partition.

> BASEOS_UPDATING should take precedence over ZDEVICE_STATE_REBOOTING.

Done